### PR TITLE
Unnecessary If/Unless

### PR DESCRIPTION
**Pull request by sodapopcan**
Here's my first pass at collapsing what I'm calling "redundant booleans," ie:

```elixir
if expr, do: true, else: false
```

I don't quite like that name and am still thinking on it but I needed something!  It only takes case of `do: true, else: false` because I realized it could be taken further but I decided that isn't worth it.  `do: false, else: true` could protentially be re-written with the `if` expression negated.  You have to chose between `!` and `not` (possibly making it configurable) and it seems like one of those things that should just be a credo warning.

I only dealt with leading comments on the `if` line though I can be more thorough.  Again, it just didn't seem worth it since people usually write these "redundant booleans" pretty absent-mindedly and can't imagine they'd be full of comments in the `do/else` bodies.  I'm also not exactly sure how to keep comments on the same line.  "Leading comments" are comments both above and after a line.  I'll study Sourceror a little more but I just wanted to get this up for review.

I'm certainly open to different ideas on the name!  I was even thinking something as dumb as `NoDoTrueElseFalse` 😅 😅 😅 

Thanks!

### CODE REVIEW

**Comment by NickNeck** [comment]
Thanks a lot. I will take a look at the weekend.

**Comment by NickNeck** [comment]
Already not looked at the code, but my suggestion for the name would be `UnnecessaryIfUnless`. As the name suggests, the implementation should also include `unless`.  

To rewrite `if expr, do: false, else: true` to `not expr` would be nice, but we can leave this for later (or never).

**Comment by sodapopcan** [comment]
All feedback has been addressed including rewriting the corresponding `unless`es.

I'm happy to squash a succinct commit if you're happy with it!

File: README.md [hunk 1]
```diff
@@ -18,24 +18,25 @@ on `sourceror`.
...
+UnnecessaryIfUnless # Corrector - Changes `if expr, do: true, else: false` to `expr`
 Warning tasks:
-Dbg               # Corrector - There should be no calls to dbg.
-IOInspect         # Corrector - There should be no calls to IO.inspect.
-TestFileExt       # Corrector - Checks the file extension of test files.
-UnusedVariable    # Corrector - Checks if unused variables occur.
+Dbg                 # Corrector - There should be no calls to dbg.
+IOInspect           # Corrector - There should be no calls to IO.inspect.
+TestFileExt         # Corrector - Checks the file extension of test files.
+UnusedVariable      # Corrector - Checks if unused variables occur.
```
Comment (sodapopcan) [inline README.md:39]: I feel a bit off about all these whitespace changes.  If you'd prefer a column cut-off with overflow I'm happy to undo.

Comment (NickNeck) [reply README.md:39]: The whitespace changes are ok.

**Review by NickNeck — APPROVED** [review]
Very nice. Thank you.

**Comment by NickNeck** [comment]
> Do you want me to squash this or does your CI handle that?

I will handle this with GitHub.

**Pull request merged**